keyboards/dmqdesign/spin/keymaps/gorbachev/keymap.c

```diff
@@ -0,0 +1,252 @@
+        switch (currentLayer) {     //break each encoder update into a switch statement for the current layer
+            case _NUMPAD:
+                if (clockwise) {
+                    tap_code(KC_DOWN);
+                } else {
+                    tap_code(KC_UP);
+                }
+                break;
+            case _RGB:
+                if (clockwise) {
+                    rgblight_increase_hue();
+                } else {
+                    rgblight_decrease_hue();
+                }
+                break;
+            case _MACRO:
+                if (clockwise) {
+                    break;
+                } else {
+                    break;
+                }
+                break;
+        }
+    } else if (index == 1) { /* Second encoder */
+        switch (currentLayer) {
+            case _NUMPAD:
+                if (clockwise) {
+                    tap_code(KC_PGDN);
+                } else {
+                    tap_code(KC_PGUP);
+                }
+                break;
+            case _RGB:
+                if (clockwise) {
+                    rgblight_increase_sat();
+                } else {
+                    rgblight_decrease_sat();
+                }
+                break;
+            case _MACRO:
+                if (clockwise) {
+                    break;
+                } else {
+                    break;
+                }
+                break;
+        }
+    } else if (index == 2) { /* Third encoder */
+        switch (currentLayer) {
+            case _NUMPAD:
+                if (clockwise) {
+                    tap_code(KC_VOLU);
+                } else {
+                    tap_code(KC_VOLD);
+                }
+                break;
+            case _RGB:
+                if (clockwise) {
+                    rgblight_increase_val();
+                } else {
+                    rgblight_decrease_val();
+                }
+                break;
+            case _MACRO:
+                if (clockwise) {
+                    break;
+                } else {
+                    break;
+                }
+                break;
+        }
+    }
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) { //This will run every time the layer is updated
+    currentLayer = get_highest_layer(state);
+
+    switch (currentLayer) {
+        case _NUMPAD:
+            setrgb(RGB_WHITE, &led[0]); //Set the top LED to white for the bottom layer
+            setrgb(0, 0, 0, &led[1]);
+            setrgb(0, 0, 0, &led[2]);
+            break;
+        case _RGB:
+            setrgb(0, 0, 0, &led[0]); //Set the middle LED to white for the middle layer
+            setrgb(RGB_WHITE, &led[1]);
+            setrgb(0, 0, 0, &led[2]);
+            break;
+        case _MACRO:
+            setrgb(0, 0, 0, &led[0]);
+            setrgb(0, 0, 0, &led[1]);
+            setrgb(RGB_WHITE, &led[2]); //Set the bottom LED to white for the top layer
+            break;
```

Comment:
This could also be accomplished by the Lighting Layer feature too.
https://docs.qmk.fm/#/feature_rgblight?id=lighting-layers

Reply:
k, this is from the encoderlayers keymap provided by the keyboard maintainer, so I'd rather keep it as is, if that's ok.

As an aside, all the other changes suggested for this PR are for code fragments directly from that same keymap, which I used as a base for my modifications. The issues you highlighted are also present there. I don't mind making the changes, obviously they make the code better, but the inconsistency between expectations from one PR to another is really confusing.

Reply:
Sounds good.